packages/core/src/shared/settings.ts

```diff
@@ -477,40 +478,40 @@ export interface ResetableMemento extends vscode.Memento {
 // at least for anything beyond primitive types.
 const settingsProps = packageJson.contributes.configuration.properties
 
-type SettingsProps = typeof settingsProps
-
-type Split<T, S extends string> = T extends `${infer L}${S}${infer R}` ? [L, ...Split<R, S>] : [T]
-type Pop<T> = T extends [...infer R, infer _] ? R : never
-type Intersection<T> = (T extends any ? (_: T) => any : never) extends (_: infer U) => any ? U : never
-
-type FromParts<T, K> = K extends [infer U, ...infer R]
-    ? [U, R] extends [string, string[]]
-        ? { [P in U & string]: FromParts<T, R> }
-        : T
-    : T
-
-type Format<T> = { [P in keyof T]: FromParts<TypeConstructor, Split<P, '.'>> }[keyof T]
-type Config = Intersection<Format<SettingsProps>>
-
-type Join<T extends string[], S extends string> = T['length'] extends 1
-    ? T[0]
-    : T extends [infer L, ...infer R]
-    ? L extends string
-        ? R extends string[]
-            ? `${L}${S}${Join<R, S>}`
-            : ''
-        : ''
-    : never
-
-type Select<T, K> = K extends [infer L, ...infer R]
-    ? L extends keyof T
-        ? R['length'] extends 0
-            ? T[L]
-            : Select<T[L], R>
-        : never
-    : never
-
-type Sections = { [P in keyof SettingsProps as Join<Pop<Split<P, '.'>>, '.'>]: Select<Config, Pop<Split<P, '.'>>> }
+// type SettingsProps = typeof settingsProps
+
+// type Split<T, S extends string> = T extends `${infer L}${S}${infer R}` ? [L, ...Split<R, S>] : [T]
+// type Pop<T> = T extends [...infer R, infer _] ? R : never
```

Comment:
oh :)

Reply:
Yeah this is an unfortunate loss, and I will remove this code instead of commenting it shortly. This code relies on the core libs package.json (that is copied to the toolkit extension). Since the standalone extensions will have divergent package.jsons, this is no longer viable.
Ideally, the core lib can supply settings classes and extensions can provide their own package.json to get type checking, but the code that access these custom settings classes would need to live in their individual packages.
This will have to be tech debt